app/api/files/list/route.ts

```diff
@@ -0,0 +1,32 @@
+import { NextResponse } from "next/server";
+import supabase from "@/lib/supabase/serverClient";
+
+export async function GET(req: Request) {
+  try {
+    const { searchParams } = new URL(req.url);
+    const ownerAccountId = searchParams.get("ownerAccountId");
+    const artistAccountId = searchParams.get("artistAccountId");
+
+    if (!ownerAccountId || !artistAccountId) {
+      return NextResponse.json({ error: "Missing ownerAccountId or artistAccountId" }, { status: 400 });
+    }
+
+    const { data, error } = await supabase
+      .from("files")
+      .select("*")
+      .eq("owner_account_id", ownerAccountId)
+      .eq("artist_account_id", artistAccountId)
+      .order("created_at", { ascending: false });
```

Comment:
Single responsibility principle
- actual: defining supabase queries in an API endpoint
- required: standalone files for supabase queries